tests/unittests/tests-crypto/tests-crypto-modes-ccm.c

```diff
@@ -17,6 +17,8 @@
 #include "crypto/modes/ccm.h"
 #include "tests-crypto.h"
 
+static const uint8_t nonce_and_len_encoding_size = 15;
```

Comment:
here and next 2 its `uint8_t` while afterwards you changed all to `size_t`. First, it doesn't make a difference (using either `uint8_t` or `size_t`) regarding binary size -> the compiler optimises that out anyway. Second, however I recommend to be consistent -> use either one but stick to *one*, and do not mix.

Reply:
I based myself of the ccm_encrypt and decrypt api:

https://github.com/RIOT-OS/RIOT/blob/b428979a1da99e4dac42688f6d3e61f7218176ed/sys/include/crypto/modes/ccm.h#L54-L57

* `mac_length` and `length_encoding` are both `uint8_t`.
* `nonce_len` which is a `size_t` has a maximum length of `15 - length_encoding` (the api documentation is wrong and saying 16 - L, I will do a PR).

Whereas the other ones `input_len`, `output_len` and `auth_data_len` can be bigger than 255 and its required to provide big data to trigger some of the bugs.

Reply:
I could replace all `uint8_t` by `size_t` anyway if you want.

Reply:
As said I checked the binary sizes for both variants, and they are the same - so the compiler likely optimises that anyway. However, IMHO its nice to be consistent if possible and not switch between types by random - so yes, I'd rather have them all `size_t` than a mix.

Reply:
Update to the documentation: https://github.com/RIOT-OS/RIOT/pull/8294